app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala

```diff
@@ -32,28 +33,41 @@ import uk.gov.hmrc.securemessage.controllers.model.common.read._
 import uk.gov.hmrc.securemessage.controllers.model.common.write._
 import uk.gov.hmrc.securemessage.controllers.utils.EnrolmentHelper._
 import uk.gov.hmrc.securemessage.controllers.utils.QueryStringValidation
-import uk.gov.hmrc.securemessage.models.core.Conversation
-import uk.gov.hmrc.securemessage.services.SecureMessageService
+import uk.gov.hmrc.securemessage.services.{ Auditing, ErrorHandling, SecureMessageService }
 import uk.gov.hmrc.time.DateTimeUtils
 
-import javax.inject.Inject
 import scala.concurrent.{ ExecutionContext, Future }
 
 @SuppressWarnings(Array("org.wartremover.warts.ImplicitParameter"))
 class SecureMessageController @Inject()(
   cc: ControllerComponents,
   val authConnector: AuthConnector,
+  override val auditConnector: AuditConnector,
   secureMessageService: SecureMessageService,
   dataTimeUtils: DateTimeUtils)(implicit ec: ExecutionContext)
-    extends BackendController(cc) with AuthorisedFunctions with QueryStringValidation with I18nSupport with Logging {
+    extends BackendController(cc) with AuthorisedFunctions with QueryStringValidation with I18nSupport
+    with ErrorHandling with Auditing {
 
   def createConversation(client: ClientName, conversationId: String): Action[JsValue] = Action.async(parse.json) {
     implicit request =>
       client match {
         case ClientName.CDCM =>
           withJsonBody[CdcmConversation] { cdcmConversation =>
-            saveConversation(
-              cdcmConversation.asConversationWithCreatedDate(client.entryName, conversationId, dataTimeUtils.now))
+            val conversation =
+              cdcmConversation.asConversationWithCreatedDate(client.entryName, conversationId, dataTimeUtils.now)
+            secureMessageService
+              .createConversation(conversation)
+              .map {
+                case Right(_) =>
+                  auditCreateConversation(EventTypes.Succeeded, conversation)
+                  Created
+                case Left(error: SecureMessageError) =>
+                  auditCreateConversation(EventTypes.Failed, conversation)
+                  handleErrors(ClientName.withName(conversation.client), conversation.id, error)
+              }
+              .recover {
+                case error: Exception => handleErrors(ClientName.withName(conversation.client), conversation.id, error)
```

Comment:
We should always wrap these kind of calls in `NonFatal`.

Reply:
@alexflav23  Exception is NonFatal, Error is fatal :) Just look at the implementation of the NonFatail :)